### ISSUE TITLE
LocaleCookieRedirect throws exception when returning file response

### ISSUE THREAD

**Issue by ridea-co** [mcamara/laravel-localization#804]
If LocaleCookieRedirect is added to the Kernel, it throws exception when returning BinaryFileResponse, which does not have "withCookie" method at line 30
![1636694792149](https://user-images.githubusercontent.com/15830174/141414718-c5112602-cff6-4a46-82f5-31c35c68fc9e.jpg)

Current work around :
1. Create a new class by copy & paste LocaleCookieRedirect
2. Check and bypass if withCookie method does not exist
3. Replace LocaleCookieRedirect with the newly created class in Kernel.php

```
        if (\count($params) > 0 && app('laravellocalization')->checkLocaleInSupportedLocales($params[0])) {
            $response = $next($request);

            if (method_exists($response, 'withCookie')) {
                return $response->withCookie(cookie()->forever('locale', $params[0]));
            } else {
                return $response;
            } // end of if

        } // end of if
```

**Comment by ijaureguialzo** [comment]
Hello @ridea-co, I was having this same issue and found that if you disable the middleware for that specific route, everything works again:

```php
Route::get('/tutor/export', 'TutorController@export')
    ->withoutMiddleware('localeCookieRedirect')
    ->name('tutor.export');
```

**Comment by iwasherefirst2** [comment]
@ridea-co  yes your right. BinaryFileResponse does not have  a withCookie method.  
Inside the else I would guess we should add a cookie using the Cookie facade instead. Cookie:forever(...).

**Comment by mgralikowski** [comment]
Another clean workaround is using config:
`'urlsIgnored' => ['*/download']`
All routes (urls) are excluded.
